x/currency/handler.go

```diff
@@ -0,0 +1,85 @@
+package currency
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+func RegisterQuery(qr weave.QueryRouter) {
+	NewTokenInfoBucket().Register("tokens", qr)
+}
+
+func RegisterRoutes(r weave.Registry, auth x.Authenticator, issuer weave.Address) {
+	r.Handle(NewTokenInfoMsg{}.Path(), NewTokenInfoHandler(auth, issuer))
+}
+
+func NewTokenInfoHandler(auth x.Authenticator, issuer weave.Address) weave.Handler {
+	return &TokenInfoHandler{
+		auth:   auth,
+		issuer: issuer,
+		bucket: NewTokenInfoBucket(),
+	}
+}
+
+type TokenInfoHandler struct {
+	auth   x.Authenticator
+	bucket *TokenInfoBucket
+	issuer weave.Address
+}
+
+func (h *TokenInfoHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	if _, err := h.validate(ctx, db, tx); err != nil {
+		return res, err
+	}
+	res.GasAllocated += newTokenInfoCost
+	return res, nil
+}
+
+const (
+	newTokenInfoCost = 100
+)
+
+func (h *TokenInfoHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.DeliverResult, error) {
+	var res weave.DeliverResult
+	msg, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return res, err
+	}
+	obj := orm.NewSimpleObj([]byte(msg.Ticker), &TokenInfo{
```

Comment:
pp: in other modules we have a `create` in the model or bucket file instead. Please be consistent.

Reply:
I disagree with this, but let's discuss it before I change it.

When reading weave and trying to understand it, it was really hard for me to realize what all those short. pubic functions are. They are weird, type fighting, public functions that assume things without explaining much. They look like wannabe ORM code. I find it much easier to understand if this code is inlined.
In addition to being in my opinion easier to understand, there is less writing and therefore less code to read and remember.

:man_shrugging:

Reply:
@husio I have the same feeling here for a while.
But, let's be consistent.
We can add another ticket to revisit this, everywhere at once.

Reply:
done :disappointed: